### PR TITLE
feat(anta.tests): Added testcase to verify vxlan1 source interface and udp port

### LINKED ISSUE #553: Add the testcase to validate Vxlan interface source interface and port

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
# Description

Added testcase to verify vxlan1 source interface and udp port.

Fixes #553 

Verifies the VXLAN source interface and UDP port.

    Expected Results:
      * success: Passes if VXLAN source interface and UDP port are correct.
      * failure: Fails if VXLAN source interface or UDP port are incorrect.
      * skipped: Skips if the Vxlan1 interface is not configured.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/custom_types.py [hunk 1]
```diff
@@ -85,6 +85,12 @@ def bgp_multiprotocol_capabilities_abbreviations(value: str) -> str:
     BeforeValidator(interface_autocomplete),
     BeforeValidator(interface_case_sensitivity),
 ]
+VxlanSrcIntf = Annotated[
+    str,
+    Field(pattern=r"^(Loopback)[0-9]+$"),
```
Comment (MaheshGSLAB) [inline anta/custom_types.py:90]: As loopback support 0-8191 so need to update the patter.

Comment (carl-baillargeon) [reply anta/custom_types.py:90]: Yes and please move this to the test itself. I doubt it will be used elsewhere.

Comment (carl-baillargeon) [reply anta/custom_types.py:90]: Oh never mind! I saw you added extra validators.

File: anta/tests/vxlan.py [hunk 2]
```diff
@@ -171,3 +173,47 @@ def test(self) -> None:
 
         if difference2:
             self.result.is_failure(f"Unexpected VTEP peer(s) on Vxlan1 interface: {sorted(difference2)}")
+
+
+class VerifyVxlan1ConnSettings(AntaTest):
+    """
+    Verifies the VXLAN source interface and UDP port.
```
Comment (carl-baillargeon) [inline anta/tests/vxlan.py:180]:
```suggestion
    Verifies the interface vxlan1 source interface and UDP port.
```

File: anta/tests/vxlan.py [hunk 3]
```diff
@@ -171,3 +173,47 @@ def test(self) -> None:
...
+
+    Expected Results:
+      * success: Passes if VXLAN source interface and UDP port are correct.
```
Comment (carl-baillargeon) [inline anta/tests/vxlan.py:183]:
```suggestion
      * success: Passes if the interface vxlan1 source interface and UDP port are correct.
```

File: anta/tests/vxlan.py [hunk 4]
```diff
@@ -171,3 +173,47 @@ def test(self) -> None:
...
+      * failure: Fails if VXLAN source interface or UDP port are incorrect.
```
Comment (carl-baillargeon) [inline anta/tests/vxlan.py:184]:
```suggestion
      * failure: Fails if the interface vxlan1 source interface or UDP port are incorrect.
```

File: anta/tests/vxlan.py [hunk 5]
```diff
@@ -171,3 +173,47 @@ def test(self) -> None:
...
+      * skipped: Skips if the Vxlan1 interface is not configured.
+    """
+
+    name = "VerifyVxlan1ConnSettings"
+    description = "Verifies the VXLAN source interface and UDP port"
```
Comment (carl-baillargeon) [inline anta/tests/vxlan.py:189]:
```suggestion
    description = "Verifies the interface vxlan1 source interface and UDP port."
```

File: anta/tests/vxlan.py [hunk 6]
```diff
@@ -171,3 +173,47 @@ def test(self) -> None:
...
+    categories = ["vxlan"]
+    commands = [AntaCommand(command="show interfaces")]
+
+    class Input(AntaTest.Input):
+        """Inputs for the VerifyVxlan1ConnSettings test."""
+
+        source_interface: VxlanSrcIntf
+        """Source loopback interface of vxlan1 interface"""
+        udp_port: int = Field(ge=1024, le=65335)
+        """UDP port used for vxlan interface"""
```
Comment (carl-baillargeon) [inline anta/tests/vxlan.py:199]:
```suggestion
        """UDP port used for vxlan1 interface"""
```

**Review by carl-baillargeon — COMMENTED** [review]
LGTM after minor changes to docstrings and comments.

Thx Mahesh :)

File: anta/custom_types.py (hunk 1 above)
Comment (MaheshGSLAB) [reply anta/custom_types.py]: Updated the custom type. Please take a look once.
@carl-baillargeon

File: anta/custom_types.py [hunk 7]
```diff
@@ -23,16 +23,20 @@ def interface_autocomplete(v: str) -> str:
 
     Supported alias:
          - `et`, `eth` will be changed to `Ethernet`
-         - `po` will be changed to `Port-Channel`"""
+         - `po` will be changed to `Port-Channel`
+         - `lo`, `lb` will be changed to `Loopback`"""
```
Comment (carl-baillargeon) [inline anta/custom_types.py:27]: `lb` is not a valid EOS syntax: 

```
DC1-LEAF1A(config-if-Et1)#int lb 0
% Invalid input
DC1-LEAF1A(config-if-Et1)#int lb0
% Invalid input
DC1-LEAF1A(config-if-Et1)#
```

**Review by carl-baillargeon — CHANGES_REQUESTED** [review]

**Review by carl-baillargeon — APPROVED** [review]
LGTM!

**Pull request merged**